src/Lean/Server/FileWorker.lean

```diff
@@ -193,9 +193,18 @@ section Initialization
     let cmdState := Elab.Command.mkState headerEnv msgLog opts
     let cmdState := { cmdState with infoState := {
       enabled := true
-      -- add dummy tree for invariant in `Snapshot.infoTree`
-      -- TODO: maybe even fill with useful stuff
-      trees := #[Elab.InfoTree.node (Elab.Info.ofCommandInfo { elaborator := `import, stx := headerStx }) #[].toPersistentArray].toPersistentArray
+      trees := #[Elab.InfoTree.context ({
+        env := headerEnv
+        fileMap := m.text
+      }) (Elab.InfoTree.node
+          (Elab.Info.ofCommandInfo { elaborator := `import, stx := headerStx })
+          (headerStx[1].getArgs.toList.map (fun importStx =>
+            Elab.InfoTree.node (Elab.Info.ofCommandInfo {
+              elaborator := Name.anonymous,
```

Comment:
The ``elaborator := `import`` part looks like a bad choice of name on my part, we should probably call that one `header` and this one `import`. If we do that, can we avoid the new big `if` in `handleDefinition` by checking if `hoverableInfoAt?` returned a `CommandInfo` with this elaborator?

Reply:
That would be the nicer solution, but `hoverableInfoAt?` doesn't seem to find the `CommandInfo` from the header.

Reply:
Ah, that's probably because `withWaitFindSnap` skips the header snapshot. I *think* lifting that restriction should be fine.